### DISCUSSION TITLE
Can't connect to wallet

### DISCUSSION THREAD

**Discussion by Mearman** [guydavis/machinaris#163] [Q&A] [answered]
Not sure if this is a recent issue as I haven't looked in a while, and I'm not sure if this isn't just me being stupid. But I can no longer connect to my wallet either through the GUI or in the container's console.

I've started and restarted the wallet and I can see the process running, but with `chia wallet show`, I still get:
```
Connection error. Check if the wallet is running at 9256. You can run the wallet via:
        chia start wallet
```

For completeness. This is my output when testing:
```
root@Polaris:/chia-blockchain# chia wallet show
Connection error. Check if the wallet is running at 9256. You can run the wallet via:
        chia start wallet
root@Polaris:/chia-blockchain# chia start wallet
chia_wallet: Already running, use `-r` to restart
chia_full_node: Already running, use `-r` to restart
root@Polaris:/chia-blockchain# chia start wallet -r
chia_wallet: stopped
chia_wallet: started
chia_full_node: stopped
chia_full_node: started
root@Polaris:/chia-blockchain# chia wallet show
Connection error. Check if the wallet is running at 9256. You can run the wallet via:
        chia start wallet
root@Polaris:/chia-blockchain# 
```

here are the corresponding logs:
(sorry for the mangled output. that's from my web view)
```
hia-blockchain/venv/bin/chia_wallet", line 33, in <module>
(load_entry_point('chia-blockchain', 'console_scripts', 'chia_wallet')())
hia-blockchain/chia/server/start_wallet.py", line 95, in main
un_service(**kwargs)
hia-blockchain/chia/server/start_service.py", line 255, in run_service
syncio.run(async_run_service(*args, **kwargs))
sr/lib/python3.9/asyncio/runners.py", line 44, in run
oop.run_until_complete(main)
sr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
uture.result()
hia-blockchain/chia/server/start_service.py", line 249, in async_run_service
wait service.run()
hia-blockchain/chia/server/start_service.py", line 171, in run
lf.start()
hia-blockchain/chia/server/start_service.py", line 135, in start
lf._node._start(**kwargs)
hia-blockchain/chia/wallet/wallet_node.py", line 159, in _start
let_state_manager = await WalletStateManager.create(
hia-blockchain/chia/wallet/wallet_state_manager.py", line 146, in create
ckchain = await WalletBlockchain.create(
hia-blockchain/chia/wallet/wallet_blockchain.py", line 116, in create
lf._load_chain_from_store()
hia-blockchain/chia/wallet/wallet_blockchain.py", line 127, in _load_chain_from_store
o_hash, sub_epoch_summaries = await self.block_store.get_peak_heights_dicts()
hia-blockchain/chia/wallet/wallet_block_store.py", line 323, in get_peak_heights_dicts
ght = hash_to_height[curr_header_hash]
```
---

Also, completely separately, how can I get plotting to automatically restart after the container boots up?

**Comment by guydavis** [comment, marked as answer]
Hi!  Please try restarting the Machinaris container.  Please look at the /mnt/user/appdata/machinaris/mainnet/log/init.log to see if there were errors on Chia service startup.  Hope this helps.

Regarding auto plot, I've updated the [wiki](https://github.com/guydavis/machinaris/wiki/Plotman#autoplot).

**Reply by Mearman** [reply]
Ok. looking at the mainnet logs I think it might be because I tried to join a pool before my mojos from the pool had confirmed and now it's getting into a bad state because it thinks it's in a pool that it hasn't actually joined yet.
Going to have a look in the config to find the pool config.

**Reply by Mearman** [reply]
I couldn't get the wallet into a good state. But after migrating my DBs and mnemonic into a new container I found I had multiple private keys, after deleting all but my main one, we're up and running again. I think the problem might be that the call to the plotnft command isn't handling multiple wallets.

**Reply by Mearman** [reply]
@guydavis I'm not going to jump the gun. what is current wallet sync time?

**Reply by guydavis** [reply]
Yes, the issue of Chia plotnft create/join calls failing if more than 1 pooling wallet (due to multiple private keys) exists was [reported ](https://github.com/guydavis/machinaris/issues/172) today.  I'll see if I can get a fix together, as the initial support definitely targets the simple case of most users, having only a single private key.  

Work around for advanced users with multiple keys and multiple pooling wallets, is to perform some of these commands themselves at the in-container CLI, thereby providing the detailed responses required to complete actions with appropriate selections.

If this response addressed your original question, please mark the Discussion as Answered.  Thanks.
